modules/extensions-mule-support/src/test/resources/models/app-as-mule-extension.json

```diff
@@ -3917,6 +3917,320 @@
       "description": "",
       "modelProperties": {},
       "kind": "operation"
+    },
+    {
+      "blocking": false,
+      "executionType": "CPU_LITE",
+      "output": {
+        "type": {
+          "format": "java",
+          "type": "String"
+        },
+        "hasDynamicType": false,
+        "description": "",
+        "modelProperties": {}
+      },
+      "outputAttributes": {
+        "type": {
+          "format": "java",
+          "type": "Void"
+        },
+        "hasDynamicType": false,
+        "description": "",
+        "modelProperties": {}
+      },
+      "transactional": false,
+      "requiresConnection": false,
+      "supportsStreaming": false,
+      "notifications": [],
+      "nestedComponents": [],
+      "errors": [],
+      "semanticTerms": [],
+      "visibility": "PUBLIC",
+      "stereotype": {
+        "type": "SALUTE_AGGRESSIVELY",
+        "namespace": "THIS",
+        "parent": {
+          "type": "PROCESSOR",
+          "namespace": "THIS",
+          "parent": {
+            "type": "PROCESSOR",
+            "namespace": "MULE"
+          }
+        }
+      },
+      "parameterGroupModels": [
+        {
+          "parameters": [
+            {
+              "type": {
+                "format": "java",
+                "type": "String"
+              },
+              "hasDynamicType": false,
+              "required": true,
+              "isConfigOverride": false,
+              "isComponentId": false,
+              "fieldValueProviderModels": [],
+              "expressionSupport": "SUPPORTED",
+              "role": "BEHAVIOUR",
+              "dslConfiguration": {
+                "allowsInlineDefinition": false,
+                "allowsReferences": false,
+                "allowTopLevelDefinition": false
+              },
```

Comment:
This is curious. Note that the [parser is generating the right thing](https://github.com/mulesoft/mule/pull/11590/files#diff-d78636ed7912dc60ef62df5315b9e7bad8e0ea3fb3636681b9964ae20ce7c65aR125) (being tested [here](https://github.com/mulesoft/mule/pull/11590/files#diff-dd22cfd0f8a8adfb635ab40b8b7dbf928823defe304a8d1c5a52b84bb270a361R95)).
Still, the `ExtensionModel` seems not to be having the right DSL Configuration. I tried to track it down and it seems it is because of [this](https://github.com/mulesoft/mule-extensions-api/blob/83c0b01fc3822b0a194a0e8f48bc625d7cf3fc21/mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/ParameterDslDeclarationEnricher.java#L96). Looks like the `ParameterDslDeclarationEnricher` is overriding the `ParameterDslConfiguration` based on the `MetadataType` and, at that point, we are declaring it as String.
Any thoughts on this @ndinu ?

Reply:
I just changed it so that it does not attempt to indicate that the parameter accepts references.
I created the task [W-11402315](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000010zRGIYA2/view) to review that decision.
But if you all agree, we can still move forward with this one as it adds value on top of what we currently have and, since it is a completely new feature, we don't risk breaking anything.